apps/web/src/components/Basenames/UsernameProfileTransferOwnershipModal/context.tsx

```diff
@@ -0,0 +1,387 @@
+        contractFunction: updateReclaim,
+      },
+      {
+        id: 'safeTransferFrom',
+        name: 'Token ownership',
+        description: 'Transfer the Basename token to this address.',
+        status: safeTransferFromStatus,
+        contractFunction: updateSafeTransferFrom,
+      },
+    ];
+  }, [
+    updateSetAddr,
+    updateSetName,
+    updateReclaim,
+    updateSafeTransferFrom,
+    reclaimStatus,
+    safeTransferFromStatus,
+    setAddrStatus,
+    setNameStatus,
+  ]);
+
+  useEffect(() => {
+    // Only when the wallet request steps is displaying
+    if (currentOwnershipStep !== OwnershipSteps.WalletRequests) return;
+
+    // Some transactions are loading / failed: return early
+    if (
+      !ownershipSettings.every((ownershipSetting) => {
+        return [
+          WriteTransactionWithReceiptStatus.Idle,
+          WriteTransactionWithReceiptStatus.Success,
+        ].includes(ownershipSetting.status);
+      })
+    )
+      return;
+
+    // Smart wallet - we can batch calls
+    if (batchCallsEnabled) {
+      updateViaBatchCalls()
+        .then()
+        .catch((error) => {
+          setCurrentOwnershipStep(OwnershipSteps.OwnershipOverview);
+          logError(error, 'Failed to update via sendCalls');
+        });
+
+      return;
+    }
+
+    // For each ownership setting / wallet transaction
+    for (const ownershipSetting of ownershipSettings) {
+      // Get the current index & previous ownership setting (if any)
+      const currentIndex = ownershipSettings.indexOf(ownershipSetting);
+      const previousOwnershipSetting = ownershipSettings[currentIndex - 1];
+
+      // If the setting transaction is Idle and either
+      // - Is the first transaction
+      // - Previous transaction is successfull
+      const canCallfunction =
+        ownershipSetting.status === WriteTransactionWithReceiptStatus.Idle &&
+        (currentIndex === 0 ||
+          previousOwnershipSetting.status === WriteTransactionWithReceiptStatus.Success);
+
+      // then call the wallet request function
+      if (canCallfunction) {
+        ownershipSetting
+          .contractFunction()
+          .then()
+          .catch((error) => {
+            logError(error, `Failed contractFunction for ${ownershipSetting.id}`);
+          });
+        break;
+      }
+    }
+  }, [batchCallsEnabled, currentOwnershipStep, logError, ownershipSettings, updateViaBatchCalls]);
+
+  const isSuccess = useMemo(
+    () =>
+      // Smart wallet: One transaction
+      batchCallsStatus === BatchCallsStatus.Success ||
+      // Other wallet: 4 Transactions are successfull
+      ownershipSettings.every(
+        (ownershipSetting) => ownershipSetting.status === WriteTransactionWithReceiptStatus.Success,
+      ),
+    [batchCallsStatus, ownershipSettings],
+  );
+
+  useEffect(() => {
+    if (isSuccess) {
+      setCurrentOwnershipStep(OwnershipSteps.Success);
+    }
+  }, [isSuccess]);
+
+  const value = useMemo(() => {
+    return {
```

Comment:
nit: can use obj return shorthand

Reply:
what's the syntax?

Reply:
`() => ({ ... })`

That said, having an explicit return can make it easier to add console logs for debugging or otherwise refactor, I'm not worried about this.